File: src/test/java/com/github/alexandrecarlton/idea/settings/applier/impl/editor/codestyle/java/JavaImportSettingsApplierTest.java
package com.github.alexandrecarlton.idea.settings.applier.impl.editor.codestyle.java;

import static org.assertj.core.api.Assertions.assertThat;

import com.github.alexandrecarlton.idea.settings.applier.api.SettingsApplier;
import com.github.alexandrecarlton.idea.settings.fixtures.IdeaSettingsTestFixture;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.ImmutableJavaImportsSettings;
import com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.JavaImportsSettings;
import com.intellij.application.options.CodeStyle;
import com.intellij.psi.codeStyle.CodeStyleSettings;
import com.intellij.psi.codeStyle.JavaCodeStyleSettings;

import org.junit.Test;

public class JavaImportSettingsApplierTest extends IdeaSettingsTestFixture {

  private CodeStyleSettings codeStyleSettings;
  private SettingsApplier<JavaImportsSettings> applier;

  @Override
  public void setUp() throws Exception {
    super.setUp();
    codeStyleSettings = CodeStyle.getSettings(project);
    applier = new JavaImportsSettingsApplier(codeStyleSettings, project);
  }

  @Override
  public void tearDown() throws Exception {
    // The normal tear-down tries to verify that Code Insight settings haven't changed,
    // so we restore this just before finishing up.
    CodeStyle.getSettings(project).copyFrom(CodeStyleSettings.getDefaults());
    super.tearDown();
  }

  @Test
  public void classCountToUseWithImportSettingsAreApplied() {
    applier.apply(ImmutableJavaImportsSettings.builder()
        .classCountToUseImportWithWildcard(555)
        .build());
    final JavaCodeStyleSettings javaCodeStyleSettings = codeStyleSettings.getCustomSettings(JavaCodeStyleSettings.class);
    assertThat(javaCodeStyleSettings.CLASS_COUNT_TO_USE_IMPORT_ON_DEMAND).isEqualTo(555);
  }

  @Test
  public void namesCountToUseWithStaticImportSettingsAreApplied() {
    applier.apply(ImmutableJavaImportsSettings.builder()
        .namesCountToUseStaticImportWithWildcard(777)
        .build());
    final JavaCodeStyleSettings javaCodeStyleSettings = codeStyleSettings.getCustomSettings(JavaCodeStyleSettings.class);
    assertThat(javaCodeStyleSettings.NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND).isEqualTo(777);
  }

}

File: applier/impl/src/test/kotlin/com/github/alexandrecarlton/idea/settings/dialog/editor/codestyle/java/imports/JavaImportSettingsApplierTest.kt
package com.github.alexandrecarlton.idea.settings.dialog.editor.codestyle.java.imports

import com.github.alexandrecarlton.idea.settings.dialog.SettingsApplier
import com.github.alexandrecarlton.idea.settings.fixtures.IdeaSettingsTestFixture
import com.intellij.psi.codeStyle.CodeStyleSettings
import org.assertj.core.api.Assertions.assertThat
import org.junit.After
import org.junit.Before
import org.junit.Test

class JavaImportSettingsApplierTest : IdeaSettingsTestFixture() {

    private lateinit var applier: SettingsApplier<JavaImportsSettings>

    @Before
    public override fun setUp() {
        applier = JavaImportsSettingsApplier(platform.javaCodeStyleSettings)
    }

    @After
    public override fun tearDown() {
        // The normal tear-down tries to verify that Code Insight settings haven't changed,
        // so we restore this just before finishing up.
        platform.codeStyleSettings.copyFrom(CodeStyleSettings.getDefaults())
    }

    @Test
    fun classCountToUseWithImportSettingsAreApplied() {
        applier.apply(JavaImportsSettings(classCountToUseImportWithWildcard = 555))
        assertThat(platform.javaCodeStyleSettings.CLASS_COUNT_TO_USE_IMPORT_ON_DEMAND).isEqualTo(555)
    }

    @Test
    fun namesCountToUseWithStaticImportSettingsAreApplied() {
        applier.apply(JavaImportsSettings(namesCountToUseStaticImportWithWildcard = 777))
        assertThat(platform.javaCodeStyleSettings.NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND).isEqualTo(777)
    }
}
